pinot-core/src/main/java/org/apache/pinot/core/minion/SegmentPurger.java

```diff
@@ -79,7 +82,7 @@ public File purgeSegment()
         return null;
       }
 
-      SegmentGeneratorConfig config = new SegmentGeneratorConfig(_tableConfig, segmentMetadata.getSchema());
+      SegmentGeneratorConfig config = new SegmentGeneratorConfig(_tableConfig, _schema);
```

Comment:
If `_schema = null`, probably it's safer to fall back to `segmentMetadata.getSchema()`?